src/components/MapForm.js

```diff
@@ -0,0 +1,304 @@
+  }
+
+  handleTimeEnd(event) {
+    const timeEnd = event.target.value
+    this.setState({ fields: { ...this.state.fields, timeEnd } })
+  }
+
+  findRegion(state) {
+    const west = ['WA', 'OR', 'CA', 'ID', 'NV', 'MT', 'WY', 'UT', 'CO', 'AZ', 'NM']
+    const midwest = ['ND', 'SD', 'NE', 'KS', 'MN', 'IA', 'MO', 'WI', 'IL', 'IN', 'MI', 'OH']
+    const south = ['TX', 'OK', 'AR', 'LA', 'MS', 'TN', 'KY', 'WV', 'MD', 'DC', 'DE', 'VA', 'NC', 'SC', 'GA', 'AL', 'FL']
+    const northeast = ['PA', 'NJ', 'RI', 'CT', 'NY', 'MA', 'VT', 'NH', 'ME']
+    const pacific = ['HI', 'AK']
+    if (south.includes(state)) {
+      return 'south'
+    }
+    else if (midwest.includes(state)) {
+      return 'midwest'
+    }
+    else if (west.includes(state)) {
+      return 'west'
+    }
+    else if (northeast.includes(state)) {
+      return 'northeast'
+    }
+    else if (pacific.includes(state)) {
+      return 'pacific'
+    }
+    return ""
+  }
+
+  updateFields(lat, lng) {
+    Geocode.fromLatLng(lat, lng)
+      .then(
+        response => {
+          let postalCode = ""
+          let stateLong = ""
+          let stateShort = ""
+          const addressComponents = response.results[0].address_components;
+          const address = response.results[0].formatted_address;
+          addressComponents.forEach(component => {
+            if (component['types'][0] === 'postal_code') {
+              postalCode = component['long_name']
+            }
+            else if (component['types'][0] === 'administrative_area_level_1') {
+              stateLong = component['long_name']
+              stateShort = component['short_name']
+            }
+          });
+          const region = this.findRegion(stateShort)
+          this.setState({
+            fields: {
+              ...this.state.fields,
+              latlng: {
+                lat,
+                lng
+              },
+              address,
+              stateLong,
+              stateShort,
+              postalCode,
+              region
+            }
+          })
+        }
+      )
+  }
+
+  async componentDidMount() {
+    const { lat, lng } = await this.getcurrentLocation();
+    await this.getAPIData();
+    this.setState(prev => ({
+      fields: {
+        ...prev.fields,
+        latlng: {
+          lat,
+          lng
+        }
+      },
+      currentLocation: {
+        lat,
+        lng
+      }
+    }));
+    this.updateFields(lat, lng)
+  }
+
+  async getAPIData() {
+    fetch('https://api.census.gov/data/timeseries/eits')
+      .then(res => res.json())
+      .then(data => {
+        this.getTables(data)
+      })
+      .catch(console.log)
+  }
+
+  getAPIGraphData = (event) => {
+    event.preventDefault()
+    fetch(`https://api.census.gov/data/timeseries/eits/${this.state.fields.tableId}?get=cell_value,data_type_code,time_slot_id,error_data,category_code,seasonally_adj&time=from+${this.state.fields.fromYear}`)
```

Comment:
Nice, this is way better than using each tables full URL!

Reply:
yeah, there was a bit of an error in your logic initially because you were using varLink, but that was there in case we end up using variables for other data sets. What we really needed to do was append the tableId to the base API URL. I intend to create a baseApiUrl variable so that we don't have the URL repeated everywhere.